app/src/kscan_sideband_behaviors.c

```diff
@@ -0,0 +1,181 @@
+            }
+        }
+
+        if (data->enabled && data->callback) {
+            data->callback(ksbb, row, column, pressed);
+        }
+    }
+}
+
+static int ksbb_configure(const struct device *dev, kscan_callback_t callback) {
+    const struct ksbb_config *cfg = dev->config;
+    struct ksbb_data *data = dev->data;
+
+    data->callback = callback;
+
+#if IS_ENABLED(CONFIG_PM_DEVICE)
+    if (pm_device_wakeup_is_enabled(dev) && pm_device_wakeup_is_capable(cfg->kscan)) {
+        pm_device_wakeup_enable(cfg->kscan, true);
+    }
+#endif // IS_ENABLED(CONFIG_PM_DEVICE)
+
+    return 0;
+}
+
+static int ksbb_enable(const struct device *dev) {
+    struct ksbb_data *data = dev->data;
+    data->enabled = true;
+
+    return 0;
+}
+
+static int ksbb_disable(const struct device *dev) {
+    struct ksbb_data *data = dev->data;
+    data->enabled = false;
+
+    return 0;
+}
+
+static int ksbb_init(const struct device *dev) {
+    const struct ksbb_config *config = dev->config;
+
+    if (!device_is_ready(config->kscan)) {
+        LOG_ERR("kscan %s is not ready", config->kscan->name);
+        return -ENODEV;
+    }
+
+    kscan_config(config->kscan, &ksbb_inner_kscan_callback);
+    kscan_enable_callback(config->kscan);
+
+    return 0;
+}
+
+static const struct kscan_driver_api ksbb_api = {
+    .config = ksbb_configure,
+    .enable_callback = ksbb_enable,
+    .disable_callback = ksbb_disable,
+};
+
+#if IS_ENABLED(CONFIG_PM_DEVICE)
+
+static int ksbb_pm_action(const struct device *dev, enum pm_device_action action) {
+    switch (action) {
+    case PM_DEVICE_ACTION_SUSPEND:
+        return ksbb_disable(dev);
+    case PM_DEVICE_ACTION_RESUME:
+        return ksbb_disable(dev);
+    default:
+        return -ENOTSUP;
+    }
+}
+
+#endif // IS_ENABLED(CONFIG_PM_DEVICE)
+
+#define JUST_ONE(_id) 1
+
+#define ENTRY(e)                                                                                   \
+    {                                                                                              \
+        .row = DT_PROP(e, row), .column = DT_PROP(e, column),                                      \
+        .binding = ZMK_KEYMAP_EXTRACT_BINDING(0, e),                                               \
+    }
+
+#define KSBB_INST(n)                                                                               \
+    static struct ksbb_entry entries_##n[] = {                                                     \
+        DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(n, ENTRY, (, ))};                                    \
+    const struct ksbb_config ksbb_config_##n = {                                                   \
+        .kscan = DEVICE_DT_GET(DT_INST_PHANDLE(n, kscan)),                                         \
+        .entries = entries_##n,                                                                    \
+        .entries_len = DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(n, JUST_ONE, (+)),                    \
```

Comment:
I think this would work?
```suggestion
        .entries_len = ARRAY_SIZE(entries_##n),                    \
```

Reply:
Indeed. Simplified and pushed.